code/game/machinery/computer/buildandrepair.dm

```diff
@@ -113,9 +113,40 @@
 			if(P.tool_behaviour == TOOL_SCREWDRIVER)
 				P.play_tool_sound(src)
 				to_chat(user, "<span class='notice'>You connect the monitor.</span>")
-				var/obj/B = new circuit.build_path(loc)
-				B.setDir(dir)
-				transfer_fingerprints_to(B)
+
+				var/obj/machinery/computer/new_computer = new circuit.build_path(loc)
+				new_computer.setDir(dir)
+				transfer_fingerprints_to(new_computer)
+
+				if(istype(new_computer))
+					// Machines will init with a set of default components. Move to nullspace so we don't trigger handle_atom_del, then qdel.
+					// Finally, replace with this frame's parts.
+					if(new_computer.circuit)
+						// Move to nullspace and delete.
+						new_computer.circuit.moveToNullspace()
+						QDEL_NULL(new_computer.circuit)
+					for(var/obj/old_part in new_computer.component_parts)
```

Comment:
Are there non-obj things in `component_parts`?

Reply:
Probably not, I'll remove the type.

Reply:
Removed type and cast to atom/movable to force_move things.